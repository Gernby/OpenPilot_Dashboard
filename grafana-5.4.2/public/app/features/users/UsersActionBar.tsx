import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames/bind';
import { setUsersSearchQuery } from './state/actions';
import { getInviteesCount, getUsersSearchQuery } from './state/selectors';

export interface Props {
  searchQuery: string;
  setUsersSearchQuery: typeof setUsersSearchQuery;
  onShowInvites: () => void;
  pendingInvitesCount: number;
  canInvite: boolean;
  showInvites: boolean;
  externalUserMngLinkUrl: string;
  externalUserMngLinkName: string;
}

export class UsersActionBar extends PureComponent<Props> {
  render() {
    const {
      canInvite,
      externalUserMngLinkName,
      externalUserMngLinkUrl,
      searchQuery,
      pendingInvitesCount,
      setUsersSearchQuery,
      onShowInvites,
      showInvites,
    } = this.props;

    const pendingInvitesButtonStyle = classNames({
      btn: true,
      'toggle-btn': true,
      active: showInvites,
    });

    const usersButtonStyle = classNames({
      btn: true,
      'toggle-btn': true,
      active: !showInvites,
    });

    return (
      <div className="page-action-bar">
        <div className="gf-form gf-form--grow">
          <label className="gf-form--has-input-icon">
            <input
              type="text"
              className="gf-form-input width-20"
              value={searchQuery}
              onChange={event => setUsersSearchQuery(event.target.value)}
              placeholder="Filter by name or type"
            />
            <i className="gf-form-input-icon fa fa-search" />
          </label>
          {pendingInvitesCount > 0 && (
            <div style={{ marginLeft: '1rem' }}>
              <button className={usersButtonStyle} key="users" onClick={onShowInvites}>
                Users
              </button>
              <button className={pendingInvitesButtonStyle} onClick={onShowInvites} key="pending-invites">
                Pending Invites ({pendingInvitesCount})
              </button>
            </div>
          )}
          <div className="page-action-bar__spacer" />
          {canInvite && (
            <a className="btn btn-success" href="org/users/invite">
              <span>Invite</span>
            </a>
          )}
          {externalUserMngLinkUrl && (
            <a className="btn btn-success" href={externalUserMngLinkUrl} target="_blank">
              <i className="fa fa-external-link-square" /> {externalUserMngLinkName}
            </a>
          )}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    searchQuery: getUsersSearchQuery(state.users),
    pendingInvitesCount: getInviteesCount(state.users),
    externalUserMngLinkName: state.users.externalUserMngLinkName,
    externalUserMngLinkUrl: state.users.externalUserMngLinkUrl,
    canInvite: state.users.canInvite,
  };
}

const mapDispatchToProps = {
  setUsersSearchQuery,
};

export default connect(mapStateToProps, mapDispatchToProps)(UsersActionBar);
