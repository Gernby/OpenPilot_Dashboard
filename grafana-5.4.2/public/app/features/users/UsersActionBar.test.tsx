import React from 'react';
import { shallow } from 'enzyme';
import { UsersActionBar, Props } from './UsersActionBar';

const setup = (propOverrides?: object) => {
  const props: Props = {
    searchQuery: '',
    setUsersSearchQuery: jest.fn(),
    onShowInvites: jest.fn(),
    pendingInvitesCount: 0,
    canInvite: false,
    externalUserMngLinkUrl: '',
    externalUserMngLinkName: '',
    showInvites: false,
  };

  Object.assign(props, propOverrides);

  return shallow(<UsersActionBar {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render pending invites button', () => {
    const wrapper = setup({
      pendingInvitesCount: 5,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should show invite button', () => {
    const wrapper = setup({
      canInvite: true,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should show external user management button', () => {
    const wrapper = setup({
      externalUserMngLinkUrl: 'some/url',
    });

    expect(wrapper).toMatchSnapshot();
  });
});
