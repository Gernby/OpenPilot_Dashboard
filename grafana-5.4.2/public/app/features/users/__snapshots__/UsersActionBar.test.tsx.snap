// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="page-action-bar"
>
  <div
    className="gf-form gf-form--grow"
  >
    <label
      className="gf-form--has-input-icon"
    >
      <input
        className="gf-form-input width-20"
        onChange={[Function]}
        placeholder="Filter by name or type"
        type="text"
        value=""
      />
      <i
        className="gf-form-input-icon fa fa-search"
      />
    </label>
    <div
      className="page-action-bar__spacer"
    />
  </div>
</div>
`;

exports[`Render should render pending invites button 1`] = `
<div
  className="page-action-bar"
>
  <div
    className="gf-form gf-form--grow"
  >
    <label
      className="gf-form--has-input-icon"
    >
      <input
        className="gf-form-input width-20"
        onChange={[Function]}
        placeholder="Filter by name or type"
        type="text"
        value=""
      />
      <i
        className="gf-form-input-icon fa fa-search"
      />
    </label>
    <div
      style={
        Object {
          "marginLeft": "1rem",
        }
      }
    >
      <button
        className="btn toggle-btn active"
        key="users"
        onClick={[MockFunction]}
      >
        Users
      </button>
      <button
        className="btn toggle-btn"
        key="pending-invites"
        onClick={[MockFunction]}
      >
        Pending Invites (
        5
        )
      </button>
    </div>
    <div
      className="page-action-bar__spacer"
    />
  </div>
</div>
`;

exports[`Render should show external user management button 1`] = `
<div
  className="page-action-bar"
>
  <div
    className="gf-form gf-form--grow"
  >
    <label
      className="gf-form--has-input-icon"
    >
      <input
        className="gf-form-input width-20"
        onChange={[Function]}
        placeholder="Filter by name or type"
        type="text"
        value=""
      />
      <i
        className="gf-form-input-icon fa fa-search"
      />
    </label>
    <div
      className="page-action-bar__spacer"
    />
    <a
      className="btn btn-success"
      href="some/url"
      target="_blank"
    >
      <i
        className="fa fa-external-link-square"
      />
       
    </a>
  </div>
</div>
`;

exports[`Render should show invite button 1`] = `
<div
  className="page-action-bar"
>
  <div
    className="gf-form gf-form--grow"
  >
    <label
      className="gf-form--has-input-icon"
    >
      <input
        className="gf-form-input width-20"
        onChange={[Function]}
        placeholder="Filter by name or type"
        type="text"
        value=""
      />
      <i
        className="gf-form-input-icon fa fa-search"
      />
    </label>
    <div
      className="page-action-bar__spacer"
    />
    <a
      className="btn btn-success"
      href="org/users/invite"
    >
      <span>
        Invite
      </span>
    </a>
  </div>
</div>
`;
